core/src/main/java/io/grpc/LoadBalancer2.java

```diff
@@ -163,6 +177,18 @@ public abstract void handleSubchannelState(
 
   /**
    * A balancing decision made by {@link SubchannelPicker} for an RPC.
+   *
+   * <p>The outcome of the decision will be one of the following:
+   * <ul>
+   *   <li>Proceed: if a Subchannel is provided via {@link #withSubchannel}, and is in READY state
+   *   when the RPC tries to start on it, the RPC will proceed on that Subchannel.</li>
+   *   <li>Error: if an error is provided via {@link #withError}, and the RPC is not wait-for-ready
+   *   (i.e., {@link CallOptions#withWaitForReady} was not called), the RPC will fail immediately
+   *   with the given error.</li>
+   *   <li>Buffer: in all other cases, the RPC will be buffered in the Channel, until next picker is
```

Comment:
I don't know what "caller" refers to.

If you are talking about the picker, because of the race between 1) a picked Subchannel being used for the RPC and 2) the Subchannel changing its state, the picker doesn't have a reliable way, and doesn't need, to decide or know about the outcome of a PickResult. As long as the LoadBalancer implementation follow the guidelines, it will work fine.

If you are talking about Channel Impl, which calls `pickSubchannel()`, it doesn't check Subchannel's state either. It calls `obtainActiveTransport()` on the internal implementation of the Subchannel, which return non-null only when the state is READY.